### PR TITLE
Frontend Status Badges

### PR DESCRIPTION
**Pull request by JVickery-TBS**
feat(templates): status badges;

- Added status badges.

As promised back in February (here: https://github.com/ckan/ckanext-xloader/pull/200#issuecomment-1928598445), this adds frontend badges stating the status of the xloadering and availability in the datastore.

Threw in the French (at least French Canadian) badges as well.

### CODE REVIEW

File: ckanext/xloader/helpers.py [hunk 1]
```diff
@@ -42,3 +44,80 @@ def is_resource_supported_by_xloader(res_dict, check_access=True):
...
+        'running': toolkit._('Loading data into DataStore'),
+        'complete': toolkit._('Data loaded into DataStore'),
+        'error': toolkit._('Failed to load data into DataStore'),
+        'active': toolkit._('Data available in DataStore'),
+        'inactive': toolkit._('Resource not active in DataStore'),
+        'unknown': toolkit._('DataStore status unknown'),
+    }
+    debug_level_statuses = ['complete', 'active', 'inactive', 'unknown']
+
+    if status in debug_level_statuses and not toolkit.asbool(toolkit.config.get('ckanext.xloader.debug_badges', False)):
```
Comment (ThrawnCA) [inline ckanext/xloader/helpers.py:99]: I would prefer to reverse this, checking a list of statuses that can be shown regardless, then if the current status isn't on that list, require the debug flag.

Comment (JVickery-TBS) [reply ckanext/xloader/helpers.py]: @ThrawnCA okay lemme know if that is better (and if I even did what you actually asked for haha). Thanks!

File: ckanext/xloader/helpers.py [hunk 2]
```diff
@@ -42,3 +44,82 @@ def is_resource_supported_by_xloader(res_dict, check_access=True):
...
+
+    is_datastore_active = resource.get('datastore_active', False)
+
+    try:
+        xloader_job = toolkit.get_action("xloader_status")({'ignore_auth': True},
+                                                           {"resource_id": resource.get('id')})
+    except toolkit.ObjectNotFound:
+        xloader_job = {}
+
+    if xloader_job.get('status') == 'complete':
```
Comment (ThrawnCA) [inline ckanext/xloader/helpers.py:69]: Could refactor this to extract `xloader_job.get('status')` to a variable at the start.

Comment (duttonw) [reply ckanext/xloader/helpers.py:69]: Is this required or just a nice code clean up @ThrawnCA .

Comment (ThrawnCA) [reply ckanext/xloader/helpers.py:69]: Just cleanup.

**Review by duttonw — COMMENTED** [review]
Readme needs to be updated with the new config options:

ckanext.xloader.show_badges : boolean (default True)
ckanext.xloader.debug_badges : boolean (default False)

**Comment by duttonw** [comment]
running it via qld-gov-au data.qld.gov.au site for end to end testing via https://github.com/qld-gov-au/ckanext-xloader/pull/108

**Comment by duttonw** [comment]
Hi @JVickery-TBS ,

We tried this in dev.data.qld.gov.au and we already run into a but, a i18n bug ;)

Carl @ThrawnCA has raised a discussion point on having better 404 handling if a i18n file is not found (use default)
https://github.com/ckan/ckan/discussions/8487

https://dev.data.qld.gov.au/static/badges/en_AU/datastore-running.svg
but
https://dev.data.qld.gov.au/static/badges/en/datastore-running.svg 
exists

My suggestion is to ditch the images and use i18n in text form only. I.e. borrow from the validations plugin.

from
```
<a href="/dataset/7795b370-eef5-4808-9844-812d7c6b302a/resource_data/550bcdfe-3ced-48a0-abcb-65d7e74c91ea" class="loader-badge"><img src="/static/badges/en_AU/datastore-running.svg" alt="Loading data into DataStore" title="October 16, 2024, 13:37 (AEST)"></a>
```
to
```<a href="/dataset/7795b370-eef5-4808-9844-812d7c6b302a/resource_data/550bcdfe-3ced-48a0-abcb-65d7e74c91ea" class="validation-badge" title="Loading data into DataStore: October 16, 2024, 13:38 (AEST)" hover=""><span class="prefix">Datastore</span><span class="status success">Running</span></a>```

image

What do you think?

I'll see if I can get round to this update as what you have is very nice. (if you don't beat me to it).


Regards

@duttonw 

p.s. I've just built a pypi github actions deployment pipeline (still needs improvements) but its now on ckna/ckanext-xloader :) You can read a bit more about it here: https://github.com/ckan/ckanext-xloader/issues/229

**Review by duttonw — APPROVED** [review]

**Pull request merged**